services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/config/JdbcProperties.java

```diff
@@ -213,6 +214,11 @@ public static JDBCClient dataSource(final Vertx vertx, final JdbcProperties data
                                 .principal(username)
                                 .credential(password)));
 
-        return JDBCClient.create(vertx, new DataSource(configuration.get()));
+        return JDBCPool.pool(vertx, new DataSource(configuration.get()),
+                new JsonObject()
```

Comment:
Yes that is a bit annoying thing to add here and I'd prefer not to do it too. But that is to avoid nullptr checks from triggering in Vertx code. I think this `JDBCPool.java` version is very close to what Hono has in use currently too;

https://github.com/vert-x3/vertx-jdbc-client/blob/b3f337fccc9975b70da673cf06771a3fd6792d72/src/main/java/io/vertx/jdbcclient/JDBCPool.java

If you look at line 101 static `pool` method, where this call ends too, it has some `Objects.requireNonNull` checks for this JsonObject contents and those trigger in Hono case. Those values are not used anywhere to my best knowledge - they just need to be non-null  :/

`maxPoolSize` is quite possibly used within Vertx `JDBCClientImpl` class for example and that is set to actual value for that reason. It was bit confusing task to try to get custom `DataSource` hooked into this `JDBCPool` currently and it seems like Vertx is changing on this part in later versions too.

Reply:
To me it looks like "jdbcUrl", "username" and "database" are used in the tracing context (line 118 in the link you provided). Therefor I would say it makes sense to set them to real values. Do you know where "maxPoolSize" is used? I had a quick look and could not find it being used anywhere.

Reply:
Yes true, I managed to overlook those uses somehow and I put proper values to those now then. For the `maxPoolSize` I've seen it defaulting to value `-1` (unlimited?) only and I am not too sure at all is it used and if it is how it is used within Vertx. DataSource default value should do connection limiting and I've been only puzzled about it should I put proper value here or leave the default behavior..

The default `maxPoolSize` value originates from this implementation which is in use currently from the `JDBCPool.pool`call;  https://github.com/vert-x3/vertx-jdbc-client/blob/4.4.9/src/main/java/io/vertx/ext/jdbc/spi/DataSourceProvider.java#L115